src/main/java/io/logz/jmx2graphite/Jmx2GraphiteJavaAgent.java

```diff
@@ -0,0 +1,99 @@
+package io.logz.jmx2graphite;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.instrument.Instrumentation;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by roiravhon on 6/6/16.
+ */
+public class Jmx2GraphiteJavaAgent {
+
+    private static final Logger logger = LoggerFactory.getLogger(Jmx2GraphiteJavaAgent.class);
+
+    public static void premain(String agentArgument, Instrumentation instrumentation) {
+
+        logger.info("Got agentArgument string: {}", agentArgument);
+
+        Map<String, String> configurationMap = getConfigurationMap(agentArgument);
+
+        if (configurationMap.get(getArgumentConfigurationRepresentation("GRAPHITE_HOSTNAME")) == null) {
+            throw new IllegalConfiguration("GRAPHITE_HOSTNAME must be one of the arguments");
+        }
+        if (configurationMap.get(getArgumentConfigurationRepresentation("SERVICE_NAME")) == null) {
+            throw new IllegalConfiguration("SERVICE_NAME must be one of the arguments");
+        }
+
+        Config userConfig = ConfigFactory.parseMap(configurationMap);
+        Config fileConfig = ConfigFactory.load("javaagent.conf");
+
+        // Merge the two configurations
+        Config finalConfig = userConfig.withFallback(fileConfig);
+
+        Jmx2GraphiteConfiguration jmx2GraphiteConfiguration = new Jmx2GraphiteConfiguration(finalConfig,
+                Jmx2GraphiteConfiguration.MetricClientType.MBEAN_PLATFORM);
+
+        Jmx2Graphite main = new Jmx2Graphite(jmx2GraphiteConfiguration);
+        logger.info("Initiated new java agent based Jmx2Graphite instance");
+
+        try {
+            main.run();
+
+          // Catching anything, because if we throw exception here, it will stop the main thread as well.
+        } catch (Throwable e) {
+            logger.error("Swallowed exception, not throwing anything but stop working.", e);
```

Comment:
logger.fatal("Stopping jmx2graphite Java Agent due to unexpected exception: "+e.getMessage(), e);